CONTRIBUTING.md

```diff
@@ -1,41 +1,53 @@
+### Use Data Generators for Blockstates and Models where possible
+To save on time and energy, most blocks and items have generated files where they are output to src/generated. Unless the block or item has a unique model that would take more time to write a generator for, please use the Data Generators provided in src/main/java/androsa/gaiadimension/data/.
 
-### Regarding JSONS, use Forge Blockstates
-If you are submitting a PR for a block alongside a JSON, please be sure to use the Forge Blockstate. These reduce on file count and is the desired format. Unless the model does not exist for the block, try to use the existing models from Minecraft.
+### Only make 16x16 pixel textures
+This might be a mod, but it needs to stay true to Minecraft's art style. Do not exceed 16x16 unless specified by the author.
 
-### Regarding Textures, make sure they are 16*16 pixels
-This might be a mod, but it needs to stay true to Minecraft's art style. Do not exceed 16*16 unless specified by the author.
-
-### Regarding Translations, make sure that it is sound quality
-I cannot understand languages outside of English, so if you are ready to have your lang file pulled, please ping the author that you are ready to do so. Lang files should also be formatted as `xx_xx`, as it is the current format.
 
 ### It needs to look good
-This is a no-brainer. To filter out random "contributors" from the mod, the texture needs at least *some* effort put into it. Try not to use third-party editors like Novaskin, as they are limited in tools. Recommended tools are Paint.NET, GIMP and Photoshop.
+This is a no-brainer. To filter out random "contributors" from the mod, the texture needs at least *some* effort put into it. Try not to use third-party editors, such as [Novaskin](https://novaskin.me/), as they are limited in tools. Recommended tools are [Paint.NET](https://www.getpaint.net/), [GIMP](https://www.gimp.org/) and [Photoshop](https://photoshop.com/).
+
+## Localization and Translating
+Most contributors cannot understand languages outside of English, so if you are ready to have your `.json` file pulled, please notify the author that you are ready to do so.
+
+### Naming conventions
+`.json` language files should be formatted as `xx_xx.json`, as it is the current and recommended format. (Example: `en_us.json`)
+
+## Other Assets Not Yet Covered By This Document
+
+### It needs to work
+You can submit a PR that's got a solution to a problem or a new feature, but it needs to build. This can be checked by submitting the PR's and looking at the mergability. This repo uses [Travis-CI](https://travis-ci.org/), so builds are often checked.
+That being said, it also needs to be functional in-game. You could submit a PR based on the grounds that it works, but thorough testing is required to make sure that it doesn't break any other feature.
```

Comment:
This subsection "It needs to work" is a repeat of an above section with the same name.